core/src/ParaGridIO.cpp

```diff
@@ -83,6 +83,10 @@ ModelState ParaGridIO::getModelState(const std::string& filePath)
 {
     ModelState state;
 
+    // TODO: Do this once rather than with every read
+    ModelMetadata& metadata = ModelMetadata::getInstance();
+    metadata.setDimensionsFromFile(filePath);
```

Comment:
Question for reviewer: Do we really want to set dimensions from file in every call to `getModelState`, as in the existing implementation? Would it not be better to do the following?
```suggestion
    ModelMetadata& metadata = ModelMetadata::getInstance();
    if (filePath /= metadata.initialFileName) {
      metadata.setDimensionsFromFile(filePath);
    }
```

Reply:
How many calls to `getModelState()` are expected? In each model run, there should only be one call to `SomethingGrid::getModelState()` and within that one call to `SomethingGridIO::getModelState()` (where `SomethingGrid` is the name of the `IStructure`implementing class). There are a few more examples around the code base for testing, but I believe that is all.

Reply:
Fair point, thanks. I removed the TODO in https://github.com/nextsimhub/nextsimdg/pull/974/commits/17ea449adb17310a1cb1f631a3806b67204d8953 and will merge once tests pass.